src/analysis/lattices/lift.h

```diff
@@ -0,0 +1,79 @@
+/*
+ * Copyright 2023 WebAssembly Community Group participants
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#ifndef wasm_analysis_lattices_lift_h
+#define wasm_analysis_lattices_lift_h
+
+#include <optional>
+#include <utility>
+
+#include "../lattice.h"
+#include "bool.h"
+
+namespace wasm::analysis {
+
+template<Lattice L> struct Lift {
+  struct Element : std::optional<typename L::Element> {
+    bool isBottom() const noexcept { return !this->has_value(); }
```

Comment:
How should I read this `has_value`? Do we use the empty optional as the bottom perhaps?

Reply:
Yes, exactly. I'll add a comment.